TTTAttributedLabel/TTTAttributedLabel.m

```diff
@@ -1904,3 +1915,50 @@ - (id)initWithCoder:(NSCoder *)aDecoder {
 }
 
 @end
+
+#pragma mark - 
+
+static inline CGColorRef CGColorRefFromColor(id color) {
+    return [color isKindOfClass:[UIColor class]] ? [color CGColor] : (__bridge CGColorRef)color;
+}
+
+static inline CTFontRef ctFontRefFromUIFont(UIFont * font) {
+    CTFontRef ctfont = CTFontCreateWithName((__bridge CFStringRef)font.fontName, font.pointSize, NULL);
+    return CFAutorelease(ctfont);
+}
+
+static inline NSDictionary * convertNSAttributedStringAttributesToCTAttributes(NSDictionary *attributes) {
+    NSMutableDictionary *mutableAttributes = [NSMutableDictionary dictionary];
+    
+    NSDictionary *NSToCTAttributeNamesMap = @{
+        NSFontAttributeName:            (NSString *)kCTFontAttributeName,
+        NSBackgroundColorAttributeName: (NSString *)kTTTBackgroundFillColorAttributeName,
+        NSForegroundColorAttributeName: (NSString *)kCTForegroundColorAttributeName,
+        NSUnderlineColorAttributeName:  (NSString *)kCTUnderlineColorAttributeName,
+        NSUnderlineStyleAttributeName:  (NSString *)kCTUnderlineStyleAttributeName,
+        NSStrokeWidthAttributeName:     (NSString *)kCTStrokeWidthAttributeName,
+        NSStrokeColorAttributeName:     (NSString *)kCTStrokeWidthAttributeName,
+        NSKernAttributeName:            (NSString *)kCTKernAttributeName,
+        NSLigatureAttributeName:        (NSString *)kCTLigatureAttributeName
+    };
+    
+    for (id originalKey in attributes) {
+        id key   = originalKey;
+        id value = attributes[key];
```

Comment:
Subscripting won't work in iOS 4. We could use `enumerateKeysAndObjectsUsingBlock:` instead